storage/innobase/dict/dict0dict.cc

```diff
@@ -7486,8 +7488,18 @@ ONLINE_TO_KEYRING*
 @return true if innodb_encrypt_tables is equal to ON or FORCE or
 ONLINE_TO_KEYRING* */
 static bool dict_should_be_keyring_encrypted() {
-  return srv_default_table_encryption == DEFAULT_TABLE_ENC_ON ||
-         srv_default_table_encryption == DEFAULT_TABLE_ENC_ONLINE_TO_KEYRING;
+  /* We cannot use srv_default_encryption here because it is
```

Comment:
Good question. Is the persist framework not available before InnoDB initialization?

Reply:
Yes, it is initialized later. I've had an issue with this and redo log encryption (persistent variables were set too late for that), that's why I disabled persisting it. I'm not 100% sure, but based on the comment you wrote here, this could be an issue here too.

Reply:
Then it looks like, there will be issue here too. I will test and file a bug or may be ask @mohitj1988 to do ;)

Reply:
I just realized that default_table_encryption to encrypt mysql.ibd is only at bootstrap time. So there shouldn't be issue

Reply:
subsequent restart do not depend on the variable and they depend on mysql.ibd page 0

Reply:
Okay, that shouldn't be an issue then.

Reply:
Great! will tick this issue as resolved..